### ISSUE TITLE
xYOLO: A Model For Real-Time Object DetectionIn Humanoid Soccer On Low-End Hardware

### ISSUE THREAD

**Issue by xinsuinizhuan** [AlexeyAB/darknet#4049]
Hi everyone,
This is achieved by trading an acceptable amount of accuracy, making the network approximately 70 times faster than Tiny-YOLO.Greater inference speed-ups were also achieved on a desktop
CPU and GPU. Additionally we contribute an annotated Darknet dataset for goal and ball detection.

https://arxiv.org/pdf/1910.03159.pdf

**Comment by TaihuLight** [comment]
@xinsuinizhuan Cloud you share the source code of xYOLO?

**Comment by ZheQU-somfy** [comment]
I've created xyolo.cfg like this,
[net]
# Testing
#batch=1
#subdivisions=1
# Training
batch=64
subdivisions=8
width=256
height=256
channels=3
momentum=0.9
decay=0.0005
angle=0
saturation = 1.5
exposure = 1.5
hue=.1

learning_rate=0.001
burn_in=1000
max_batches = 40000
policy=steps
steps=128000,216000
scales=.1,.1

[convolutional]
batch_normalize=1
filters=2
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
bin_output=1
batch_normalize=1
filters=4
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
bin_output=1
batch_normalize=1
filters=8
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
bin_output=1
batch_normalize=1
filters=16
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=2

[convolutional]
bin_output=1
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[maxpool]
size=2
stride=1

[convolutional]
xnor=1
bin_output=1
batch_normalize=1
filters=128
size=3
stride=1
pad=1
activation=leaky

###########

[convolutional]
xnor=1
batch_normalize=1
filters=32
size=1
stride=1
pad=1
activation=leaky

[convolutional]
xnor=1
batch_normalize=1
filters=64
size=3
stride=1
pad=1
activation=leaky

[convolutional]
xnor=1
size=1
stride=1
pad=1
filters=75
activation=leaky


[yolo]
mask = 3,4,5
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=20
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1

[route]
layers = -4

[convolutional]
batch_normalize=1
filters=16
size=1
stride=1
pad=1
activation=leaky

[upsample]
stride=2

[route]
layers = -1, 8

[convolutional]
batch_normalize=1
filters=32
size=3
stride=1
pad=1
activation=leaky


[convolutional]
size=1
stride=1
pad=1
filters=75
activation=linear

[yolo]
mask = 0,1,2
anchors = 10,14,  23,27,  37,58,  81,82,  135,169,  344,319
classes=20
num=6
jitter=.3
ignore_thresh = .7
truth_thresh = 1
random=1
And when I lauched my trainning, I've got some segmentation erros in function binarize_weights ().
Could you please give me some help?
